core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -527,7 +536,43 @@ public boolean isTerminated() {
     return terminated;
   }
 
-  /*
+  @Override
+  public ConnectivityState getState(boolean requestConnection) {
+    // TODO: handle requestConnection. At this point, not really sure how.
+    return state;
+  }
+
+  @Override
+  public void notifyWhenStateChanged(ConnectivityState source, Runnable callback) {
```

Comment:
Ahh yes. I misunderstood it indeed. Makes sense but now my question is how many times the `Runnable` will be ran?

I assume that every time it gets triggered, the callback should be removed and if users want to continue having it registered, they should re-register it inside `Runnable`. Is that correct?